### PR TITLE
Fixes possible buildout issue

### PR DESCRIPTION
**Pull request by jweede**
For some reason calling the attrs on buildout directly in recipes can cause issues with larger custom buildouts. 
Buildout source advises against calling buildout attributes inside recipes. 
Here's a wrapper passed to jinja that redirects attribute calls to getitems instead.

### CODE REVIEW

File: brodul/recipe/template/__init__.py [hunk 1]
```diff
@@ -214,3 +214,14 @@ def _ensure_dir(self, directory):
...
+
+class SafeBuildout(object):
+
+    def __init__(self, dict_like):
+        self.dict_like = dict_like
+        self.get = dict_like.get
+        self.keys = dict_like.keys
+
+    def __getattr__(self, item):
+        return self.dict_like.get(item)
```
Comment (nsheaps) [inline brodul/recipe/template/__init__.py:227]: Maybe override `__setattr__` as well?

Comment (jweede) [inline brodul/recipe/template/__init__.py:227]: We purposely don't want templates to be able to change the internal state of buildout.

Comment (nsheaps) [inline brodul/recipe/template/__init__.py:227]: Correct. Doesn't the default functionality of `object.__setattr__` go against this, though it wouldn't affect the get methods

Comment (jweede) [inline brodul/recipe/template/__init__.py:227]: In such a scenario, we'd see the SafeBuildout instance being modified. Which is fine.

**Pull request closed without merging**